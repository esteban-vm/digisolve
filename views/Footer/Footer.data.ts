import type { NavLinkProps } from './NavLink'
import type { SocialLinkProps } from './SocialLink'

export const navLinks: NavLinkProps[] = [
  {
    id: '97e51480-e825-11ed-b24d-893619994521',
    text: 'About us',
  },
  {
    id: '9c0412a0-e825-11ed-b24d-893619994521',
    text: 'Blog',
  },
  {
    id: 'a1fc3d90-e825-11ed-b24d-893619994521',
    text: 'Help & Support',
  },
  {
    id: 'a4f929e0-e825-11ed-b24d-893619994521',
    text: 'iOS App',
  },
  {
    id: 'a7b0f6e0-e825-11ed-b24d-893619994521',
    text: 'Android App',
  },
]

export const socialLinks: SocialLinkProps[] = [
  {
    id: 'bb690ae0-8c1b-428c-9113-6e22cfb5c44b',
    text: 'Facebook',
    icon: 'facebook-f',
  },
  {
    id: '8083a63a-bd9c-4a89-b7ae-025e873529a8',
    text: 'Twitter',
    icon: 'twitter',
  },
  {
    id: '97bc93f4-ab8c-44dd-a33e-b8c5d6c8260b',
    text: 'TikTok',
    icon: 'tiktok',
  },
  {
    id: '42f20ef7-eabb-485c-8f69-4155f45623e4',
    text: 'Instagram',
    icon: 'instagram',
  },
  {
    id: '14c8210d-8242-48a5-ac7c-be9b5111b213',
    text: 'YouTube',
    icon: 'youtube',
  },
]
