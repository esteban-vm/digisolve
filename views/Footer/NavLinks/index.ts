export { default, navLinks } from './NavLinks'
