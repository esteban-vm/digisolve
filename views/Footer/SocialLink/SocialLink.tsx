import type { Component, PropsWithLink } from '@/types'
import Link from 'next/link'
import { Icon, type IconName } from '@/components'
import { styled } from '@/styles'

export type SocialLinkProps = PropsWithLink<{ text: string; icon: IconName }>

const SocialLinkComponent: Component<SocialLinkProps> = ({ text, link = '#', icon, ...rest }) => {
  return (
    <li {...rest}>
      <Link href={link} target='_blank' rel='noopener noreferrer' title={text}>
        <Icon name={icon} size={1.5} className='icon' />
      </Link>
    </li>
  )
}

const SocialLink = styled(SocialLinkComponent)`
  margin-left: 20px;

  .icon {
    color: inherit;
  }
`

export default SocialLink
