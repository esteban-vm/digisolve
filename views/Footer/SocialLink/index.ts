export { default /* type SocialLinkProps */ } from './SocialLink'
