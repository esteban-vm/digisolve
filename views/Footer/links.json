{
  "navLinks": [
    { "text": "About us" },
    { "text": "Blog" },
    { "text": "Help & Support" },
    { "text": "iOS App" },
    { "text": "Android App" }
  ],
  "socialLinks": [
    { "icon": ["fab", "facebook-f"], "title": "Facebook" },
    { "icon": ["fab", "twitter"], "title": "Twitter" },
    { "icon": ["fab", "google-plus-g"], "title": "Google Plus" },
    { "icon": ["fab", "instagram"], "title": "Instagram" },
    { "icon": ["fab", "youtube"], "title": "YouTube" }
  ]
}
