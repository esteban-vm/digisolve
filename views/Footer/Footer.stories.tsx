import type { Meta } from '@storybook/react'
import Footer from './Footer'

export default {
  component: Footer,
  title: 'Views/Footer',
} as Meta<typeof Footer>

export const FooterView = () => <Footer />
