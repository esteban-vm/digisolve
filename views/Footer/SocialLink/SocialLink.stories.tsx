import type { Meta } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import { socialLinks } from '../Footer.data'
import SocialLink from './SocialLink'

const Wrapper = styled.div`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ul {
    float: right;

    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }

      :first-of-type {
        a:hover {
          color: var(--color-facebook);
        }
      }

      :nth-of-type(2) {
        a:hover {
          color: var(--color-twitter);
        }
      }

      :nth-of-type(3) {
        a:hover {
          color: var(--color-tiktok);
        }
      }

      :nth-of-type(4) {
        a:hover {
          color: var(--color-instagram);
        }
      }

      :last-of-type {
        a:hover {
          color: var(--color-youtube);
        }
      }
    }
  }
`

export default {
  component: SocialLink,
  title: 'Views/Footer/Social Links',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isHalf />
            <Grid.Col isHalf>
              <Story {...context} />
            </Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    viewport: {
      disable: true,
    },
  },
} as Meta<typeof SocialLink>

export const Example = () => {
  return (
    <ul>
      {socialLinks.map((link) => (
        <SocialLink key={link.id} {...link} />
      ))}
    </ul>
  )
}
