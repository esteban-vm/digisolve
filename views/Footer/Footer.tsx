import type { Component } from '@/types'
import { Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import { navLinks, socialLinks } from './Footer.data'
import NavLink from './NavLink'
import SocialLink from './SocialLink'

const FooterComponent: Component = (props) => {
  return (
    <footer {...props}>
      <Grid.Row>
        <Grid.Col isHalf>
          <ul className='nav__links'>
            {navLinks.map((link) => (
              <NavLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
        <Grid.Col isHalf>
          <ul className='social__links'>
            {socialLinks.map((link) => (
              <SocialLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
      </Grid.Row>
      <Grid.Row>
        <p>This webpage has been created for educational purposes.</p>
        <p>You are feel free to use it and customize it to make your own webpages.</p>
      </Grid.Row>
    </footer>
  )
}

const Footer = styled(FooterComponent)`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ${mediaQuery('md')} {
    padding: 30px;
  }

  ul {
    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }

    &[class$='__links'] {
      ${mediaQuery('xs')} {
        float: none;
        text-align: center;
        margin-bottom: 10px;
      }
    }
  }

  p {
    text-align: center;
    line-height: 1.5;

    ${mediaQuery('md')} {
      line-height: 1.3;
    }

    :first-of-type {
      margin-top: 50px;

      ${mediaQuery('md')} {
        margin-top: 20px;
      }
    }
  }

  .nav__links {
    float: left;
  }

  .social__links {
    float: right;
  }
`

export default Footer
