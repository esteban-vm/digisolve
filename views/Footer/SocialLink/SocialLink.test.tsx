import { render, cleanup, screen, create, axe } from '@/tests'
import { socialLinks } from '../Footer.data'
import SocialLink from './SocialLink'

describe('ðŸ§ª SOCIAL LINK:', () => {
  const [testLink] = socialLinks

  describe('render tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(
        <ul>
          <SocialLink {...testLink} />
        </ul>
      ))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    it('the icon of the link should be rendered', () => {
      const link = screen.getByRole('link', { name: testLink.text })
      const icon = screen.getByRole('img', { hidden: true })
      expect(link).toBeInTheDocument()
      expect(icon).toBeInTheDocument()
      expect(icon).toBeVisible()
      expect(link).toContainElement(icon)
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<SocialLink {...testLink} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
