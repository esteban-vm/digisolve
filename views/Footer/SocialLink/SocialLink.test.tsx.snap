// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª SOCIAL LINK: style tests: should render with correct styles 1`] = `
.emotion-0 {
  margin-left: 20px;
}

.emotion-0 .icon {
  color: inherit;
}

<li
  className="emotion-0"
  id="bb690ae0-8c1b-428c-9113-6e22cfb5c44b"
>
  <a
    href="#"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
    rel="noopener noreferrer"
    target="_blank"
    title="Facebook"
  >
    <svg
      aria-hidden={true}
      className="icon"
      color="var(--color-primary)"
      fill="currentColor"
      height="1.5rem"
      role="img"
      viewBox="0 0 320 512"
      width="1.5rem"
    >
      <path
        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
      />
    </svg>
  </a>
</li>
`;
